erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -1062,6 +1062,7 @@ def set_missing_values(source, target):
 		# set the redeem loyalty points if provided via shopping cart
 		if source.loyalty_points and source.order_type == "Shopping Cart":
 			target.redeem_loyalty_points = 1
+			target.loyalty_points = source.loyalty_points
```

Comment:
Ensure `si.loyalty_points` is not `None` and therefore LPs are applied when desired